python/lsst/daf/butler/datastores/remoteFileDatastore.py

```diff
@@ -0,0 +1,239 @@
+    config : `DatastoreConfig` or `str`
+        Configuration. A string should refer to the name of the config file.
+    bridgeManager : `DatastoreRegistryBridgeManager`
+        Object that manages the interface between `Registry` and datastores.
+    butlerRoot : `str`, optional
+        New datastore root to use to override the configuration value.
+
+    Raises
+    ------
+    ValueError
+        If root location does not exist and ``create`` is `False` in the
+        configuration.
+
+    Notes
+    -----
+    Datastore supports non-link transfer modes for file-based ingest:
+    `"move"`, `"copy"`, and `None` (no transfer).
+    """
+
+    def __init__(self, config: Union[DatastoreConfig, str],
+                 bridgeManager: DatastoreRegistryBridgeManager, butlerRoot: str = None):
+        super().__init__(config, bridgeManager, butlerRoot)
+        if not self.root.exists():
+            try:
+                self.root.mkdir()
+            except ValueError as e:
+                raise ValueError(f"Can not create datastore root '{self.root}', check permissions.") from e
+
+    def _artifact_exists(self, location: Location) -> bool:
+        """Check that an artifact exists in this datastore at the specified
+        location.
+
+        Parameters
+        ----------
+        location : `Location`
+            Expected location of the artifact associated with this datastore.
+
+        Returns
+        -------
+        exists : `bool`
+            True if the location can be found, false otherwise.
+        """
+        log.debug("Checking if resource exists: %s", location.uri)
+        return location.uri.exists()
+
+    def _delete_artifact(self, location: Location) -> None:
+        """Delete the artifact from the datastore.
+
+        Parameters
+        ----------
+        location : `Location`
+            Location of the artifact associated with this datastore.
+        """
+        log.debug("Deleting file: %s", location.uri)
+        location.uri.remove()
+        log.debug("Successfully deleted file: %s", location.uri)
+
+    def _read_artifact_into_memory(self, getInfo: DatastoreFileGetInformation,
+                                   ref: DatasetRef, isComponent: bool = False) -> Any:
+        location = getInfo.location
+
+        log.debug("Downloading data from %s", location.uri)
+        serializedDataset = location.uri.read()
+
+        storedFileInfo = getInfo.info
+        if len(serializedDataset) != storedFileInfo.file_size:
+            raise RuntimeError("Integrity failure in Datastore. "
+                               f"Size of file {location.path} ({len(serializedDataset)}) "
+                               f"does not match recorded size of {storedFileInfo.file_size}")
+
+        # format the downloaded bytes into appropriate object directly, or via
+        # tempfile (when formatter does not support to/from/Bytes). This is
+        # equivalent of PosixDatastore formatter.read try-except block.
```

Comment:
> read try-except block.
I don't know if you meant to leave this, or need to say it? (I assume the comment refers to the code block that follows it. Or, you're missing a space after the period that just precedes it.

Reply:
formatter.read is a method name not a sentence end